Find a functionally equivalent code:com.android.systemui.statusbar.SignalClusterView.apply:COMMENT
Method Modifier: private     
Comment:// Run after each indicator change.

@@ -1,6 +1,9 @@
 {
     if (mWifiGroup == null)
         return;
+    mVpn.setVisibility(mVpnVisible ? View.VISIBLE : View.GONE);
+    if (DEBUG)
+        Log.d(TAG, String.format("vpn: %s", mVpnVisible ? "VISIBLE" : "GONE"));
     if (mWifiVisible) {
         mWifi.setImageResource(mWifiStrengthId);
         mWifiGroup.setContentDescription(mWifiDescription);
@@ -24,35 +27,44 @@
     } else {
         mAirplane.setVisibility(View.GONE);
     }
-    if (mMobileVisible && mWifiVisible && mIsAirplaneMode) {
-        mSpacer.setVisibility(View.INVISIBLE);
+    if (mIsAirplaneMode && mWifiVisible) {
+        mWifiAirplaneSpacer.setVisibility(View.VISIBLE);
+    } else {
+        mWifiAirplaneSpacer.setVisibility(View.GONE);
+    }
+    if (mRoaming && mMobileVisible && mWifiVisible) {
+        mWifiSignalSpacer.setVisibility(View.VISIBLE);
     } else {
-        mSpacer.setVisibility(View.GONE);
+        mWifiSignalSpacer.setVisibility(View.GONE);
     }
+    mMobile.setPaddingRelative(mIsMobileTypeIconWide ? mWideTypeIconStartPadding : 0, 0, 0, 0);
     if (DEBUG)
         Log.d(TAG, String.format("mobile: %s sig=%d typ=%d", (mMobileVisible ? "VISIBLE" : "GONE"), mMobileStrengthId, mMobileTypeId));
-    mMobileType.setVisibility(!mWifiVisible ? View.VISIBLE : View.GONE);
+    mMobileType.setVisibility((mRoaming || mMobileTypeId != 0) ? View.VISIBLE : View.GONE);
 }

